FEXCore/Source/Utils/FileLoading.cpp

```diff
@@ -24,24 +24,43 @@ static bool LoadFileImpl(T &Data, const fextl::string &Filepath, size_t FixedSiz
   size_t FileSize{};
   if (FixedSize == 0) {
     struct stat buf;
-    if (fstat(FD, &buf) != 0) {
-      close(FD);
-      return false;
+    if (fstat(FD, &buf) == 0) {
+      FileSize = buf.st_size;
     }
-
-    FileSize = buf.st_size;
   }
   else {
     FileSize = FixedSize;
   }
 
   ssize_t Read = -1;
-  if (FileSize > 0) {
-    Data.resize(FileSize);
-    Read = pread(FD, &Data.at(0), FileSize, 0);
+  bool LoadedFile{};
+  if (FileSize) {
```

Comment:
This will handle empty files poorly. Not the most important use case, but easily avoided by changing `FileSize` to be an `std::optional<size_t>`.

Reply:
We can't distinguish empty files from memory backed files, fstat returns success for both, with st_size being zero for both.

Reply:
Oof 😩